app/assets/javascripts/to_do_items.coffee

```diff
@@ -90,24 +100,43 @@ class @TodoItem
     currentCheckbox = $("##{id}")
     currentCheckbox.prop 'checked', !currentCheckbox.prop 'checked'
 
-  showCheckAlert = (listId) ->
-    alertBox = detectList(listId).find('div.alert')
+  showAlert = (listId, typeError) ->
+    alertBox = detectList(listId).find("div.#{typeError}")
     return unless alertBox.css('display') == 'none'
     alertBox.fadeIn('slow').delay(3000).fadeOut('slow')
 
-  renderCheckError = (item) ->
+  onCheckError = (item) ->
     revertChecking item.attr('id')
-    showCheckAlert item.attr('value')
+    showAlert item.attr('value'), 'mark-error'
+
+  onMoveError = (item) ->
+    listId = item.find("input[type='checkbox']").attr('value')
+    revertMoving listId
+    showAlert listId, 'move-error'
 
   doCheckRequest = (item) ->
     $.ajax
       url: item.attr('url')
       method: 'PUT'
       error: () ->
-        renderCheckError item
+        onCheckError item
       success: (response) ->
         toggle response
 
+  changePositionRequest = (item) ->
+    id = extractId item.attr('id')
+    $.ajax
+      url: item.attr('move')
+      data: {id: id, position: item.index() + 1}
+      method: 'PUT'
+      error: () ->
+        onMoveError item
+      success: () ->
+        highlightNewOne extractId item.attr('id')
+
+  updatePosition = (item) ->
+    changePositionRequest item
```

Comment:
@vlasiak Now let's make the following adjustments into `updatePosition` method and the one that makes ajax request:
- Put any code that works with parameters and extracts their values from the item into `updatePosition`method.
- Pass only `options` parameter into method that makes ajax request. As `item` is needed inside, pass it inside `options` as well. So the `options` would contain: `id`, `position`, `url`. Another name for `options` is `params`.